eras/alonzo/impl/testlib/Test/Cardano/Ledger/Alonzo/TreeDiff.hs

```diff
@@ -172,3 +176,17 @@ instance
         , ("pwcExUnits", toExpr pwcExUnits)
         , ("pwcCostModel", toExpr pwcCostModel)
         ]
+
+deriving instance Generic (TransactionScriptFailure era)
+
+instance
+  ( Era era
+  , ToExpr (TxCert era)
+  , ToExpr (ContextError era)
+  , ToExpr (PlutusScript era)
+  , ToExpr (PlutusPurpose AsIx era)
+  , ToExpr (PlutusPurpose AsItem era)
+  , ToExpr EvaluationError
+  , ToExpr ExBudget
+  ) =>
+  ToExpr (TransactionScriptFailure era)
```

Comment:
So, why did you need this instance?

Currently it is broken, so you could not have used it in any of the tests, therefore I suspect it can be removed.

On the other hand if we really do need it then we need to define orphan `ToExpr` instances for `EvaluationError` and `ExBudget`. However, those would have to be placed into `cardano-ledgre-core:testlib`

Reply:
I originally added it so I could use `ansiExpr` instead of `viaShow` [here](https://github.com/IntersectMBO/cardano-ledger/blob/e34138b741bb5bace2949b19b74ed2aa3cd83ff1/eras/alonzo/impl/testlib/Test/Cardano/Ledger/Alonzo/ImpTest.hs#L213). However, the same module then needed `ToExpr (PlutusPurpose AsIx era)` so I gave up and just used `viaShow` (but didn't take out the instance right away in case I wanted to go back to it).